test/functional/test_framework/test_framework.py

```diff
@@ -1415,27 +1410,18 @@ def update_registrar(self, node: TestNode, submit: bool, pubKeyOperator: Optiona
 
         return ret
 
-    def update_service(self, node: TestNode, submit: bool, coreP2PAddrs: Union[str, List[str], None] = None, platform_node_id: Optional[str] = None, platform_p2p_port: Optional[int] = None,
-                       platform_http_port: Optional[int] = None, address_operator: Optional[str] = None, fundsAddr: Optional[str] = None,
-                       expected_assert_code: Optional[int] = None, expected_assert_msg: Optional[str] = None) -> Optional[str]:
-        if (expected_assert_code and not expected_assert_msg) or (not expected_assert_code and expected_assert_msg):
-            raise AssertionError("Intending to use assert_raises_rpc_error() but didn't specify code and message")
+    def update_service(self, node: TestNode, submit: bool, addrs_core_p2p: Union[str, List[str], None] = None, platform_node_id: Optional[str] = None,
+                       addrs_platform_p2p: Union[int, str, List[str], None] = None, addrs_platform_https: Union[int, str, List[str], None] = None,
```

Comment:
is there any benefits to have type `Union[int, str, List[str], None]` specified?
It makes code a messy but it seems as almost everything will fit this Union